src/units/abilities.cpp

```diff
@@ -1518,6 +1518,19 @@ bool attack_type::has_special_or_ability(const std::string& special, bool specia
 }
 //end of emulate weapon special functions.
 
+//this function is used for check specials wo can be removed in [set_specials] when [filter_special] matched
+void attack_type::remove_special_by_attribute(const config& filter)
+{
+	config::all_children_iterator i = specials_.ordered_begin();
+	while (i != specials_.ordered_end()) {
+		if(self_ && (*self_).ability_matches_filter(i->cfg, i->key, filter , true)) {
```

Comment:
Why does this need a new `special_only` parameter? What happens if that isn't added?

I'm asking because I'm wondering why this needs to make other code more complex.

Reply:
i removed that because don't work with (*self_).ability_matches_filter, i create attac_type equivalent